core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -290,14 +290,10 @@ public String routing() {
     }
 
     /**
-     * Sets the parent id of this document. If routing is not set, automatically set it as the
-     * routing as well.
+     * Sets the parent id of this document.
      */
     public IndexRequest parent(String parent) {
         this.parent = parent;
-        if (routing == null) {
-            routing = parent;
-        }
```

Comment:
you should change the writeTo method to take care of nodes on older versions like in DeleteRequest